### PR TITLE
New multicurrency sync rates

### PR DESCRIPTION
**Pull request by aconits**
Hi @eldy 
I work agin on multicurrency admin. 
Now we can synchronize all rates with currencylayer app (create if not exists or update rate).

:smiley:

### CODE REVIEW

File: htdocs/admin/multicurrency.php [hunk 1]
```diff
@@ -41,10 +42,17 @@
...
-
+$sync_response = GETPOST('sync_response');
 /*
  * Actions
  */
+if (!empty($sync_response))
+{
+	$sync_response = json_decode($sync_response);
+	MultiCurrency::syncRates($sync_response);
+	exit;
```
Comment (eldy) [inline htdocs/admin/multicurrency.php]: This exit looks strange to me. No output to user ?

Comment (aconits) [inline htdocs/admin/multicurrency.php]: No, the synchronize work from JS, if we got error from the webservice I notice it by JS event or if all is right I refresh by JS

**Comment by aconits** [comment]
@eldy 
ready to merge

File: htdocs/admin/multicurrency.php (hunk 1 above)
Comment (eldy) [inline htdocs/admin/multicurrency.php]: The exit is still wrong here.
This means the code behind MultiCurrency::syncRates($sync_response); is just a code to run an upate of database from javascript main page. If yes, you must create another dedicate page called   ajax/updaterates.php for example to receive the parameter "sync_response" and make the silent update.

**Comment by aconits** [comment]
@eldy 
So, I moved the code to make a silent update. This commit is right ?

**Comment by eldy** [comment]
No, this is worse. Now we get a blank page.
The code that is the ajax asynchrone action must just be moved into another file call .../ajax/dotheajaxaction.php
All php files that are not pages but Ajax/api call must be stored into external php files into an ajax subdir (see htdocs/core/ajax/ dir for other cases)

**Comment by aconits** [comment]
@eldy 
I'm totally lost, should I create a file into /htdocs/core/ajax and then move all code behind MultiCurrency::syncRates into this file ?

Currently the job run like this : 
1. ajax call to get all values from the API
2. send a request by ajax to create/update rates into database
3. reload admin page

Maybe the good way is to get values from the API with ajax call then reload and do the create/update rates ?

**Comment by eldy** [comment]
Things are easy:
If a call of a php file return a json content of just a return code, it is not a page, it must be an dedicated ajax file stored into core/ajax
If this is a page call that return/output a full html screen, it is a standard page. You should never get into a standard page an "exit" just after an action. It means it does not output a html page, so it must be an ajax page.

There is no good/bad way, just rules to keep code cleaner. Calling a page just to make MultiCurrency::syncRates and ending with exit is not show ing a page. So you just have to move code of action into a file stored into core/ajax dir.

**Comment by aconits** [comment]
@eldy 
So, I have remove the part of ajax to update the rates.
I keep the first ajax call to get rates from the API then I send the response with form.

I hope this way is correct :sweat:

**Comment by eldy** [comment]
It is enough to say I don't see critical conception default. So I validate.
But may be you removed a feature to achieve this successful PR. If you want to reintroduce, just create an issue an assign it to me and you (if you can) so we can still echange on this topic.

**Pull request merged**